docuvision/batchsendfulldir.py

```diff
@@ -0,0 +1,204 @@
+if APITOKEN is None:
+    logging.error("DOCUVISION_API_TOKEN environment variable is not set. Aborting")
+    sys.exit()
+if APIADDRESS is None:
+    logging.error("DOCUVISION_API_ADDRESS environment variable is not set. Aborting")
+    sys.exit()
+if APIADDRESS[-1] != "/": APIADDRESS+="/"
+logging.info("APIADDRESS loaded. Using {}".format(APIADDRESS))
+logging.info("APITOKEN loaded. Using {}...".format(APITOKEN[:15]))
+
+
+#expects filepath to be a Path() object
+#timeout is in seconds
+#writes the newly created 'jobid filepath' as a line in pendingjobs.docuvision file in current directory
+#returns job id (int)
+def hankai_submit_job(filepath, timeout=60):
+    with open(file, 'rb') as f:
+        raw_bytes = f.read()
+        enc_bytes = base64.encodebytes(raw_bytes)
+    enc_string = enc_bytes.decode("utf-8")
+    md5_sum = hashlib.md5(raw_bytes).hexdigest()
+    headers = {"x-api-key": APITOKEN }
+    req = {
+        "name": "customername_job_x",
+        "request": {
+            "document": {
+                "name": filepath.name,
+                "dataType": "blob",
+                "encodingType": "base64/utf-8",
+                "mimeType": filepath.suffix[1:], #removes the leading period
+                "model": "base-medrec-anesthesia", #modify this to the model you want to consume
+                "confidenceInterval": 0.7, #modify this to your liking
+                "isPerformOCR": True, #if set your result will have OCR'd words and bounding boxes
+                "sizeBytes": len(enc_bytes),
+                "md5Sum": md5_sum,
+                "data": enc_string,
+            }
+        }
+    }
+    resp = requests.request(
+        method="POST",
+        url=f"{APIADDRESS}", #docuvision/v1/tasks/7",
+        headers=headers,
+        data=json.dumps(req),
+        timeout=timeout
+    )
+    rjson = json.loads(resp.content)
+    print(resp.content)
+    req_id = rjson.get("id")
+    req_meta = rjson.get("metadata", {})
+    if 'apiKey' in req_meta.keys(): req_meta['apiKey']='hidden'
+    logging.info(f"Job posted. status={resp.status_code}. id={req_id}. file={filepath.name}. respmeta={req_meta}")
+
+    if req_id is not None: 
+        #write the job id of the newly created job to local file 
+        with open('pendingjobs.docuvision', 'a') as f:
+            f.write(f"{req_id} {filepath}\n")
+    return req_id
+
+#gets the api response for a given jobid
+def hankai_get_results(jobid):
+    logging.debug("Getting {}{}".format(APIADDRESS, jobid))
+    headers = {"x-api-key": APITOKEN }
+    resp = requests.request(
+        method="GET",
+        url=f"{APIADDRESS}{jobid}",
+        headers=headers
+    )
+    return resp
+
+#returns 1 if the job is complete (even if state is error) or 0 if not found or in progress
+def hankai_check_job_complete(resp):
+    try:
+        rjson= json.loads(resp.content)
+        if (rjson.get("state") and rjson.get("state") != "In-Progress"):
```

Comment:
If Error thats also not in progress nor completed. Error should be handled differently.i would recommend definitely not remove it but mark it as error. Sorry doing reviewc on my phone

Reply:
ok. well the job *is* complete just as an error. should it save that response as a .json file or discard it and leave no json beside the original?

Reply:
I think saving json is helpful for debugging but also somehow to indicate in the log that its an error. In other words perhaps there can be an error log? You decide but knowing that it failed and how it failed [json] will definitely help.